giflib failed to build on OS X 10.5.9

➜  ~ HOMEBREW_MAKE_JOBS=1 VERBOSE=1 brew install giflib
==> Downloading https://prdownloads.sourceforge.net/project/giflib/giflib-5.2.1.tar.gz
Already downloaded: /Users/brian/Library/Caches/Homebrew/giflib-5.2.1.tar.gz
==> Verifying giflib-5.2.1.tar.gz checksum
/usr/bin/tar xzf /Users/brian/Library/Caches/Homebrew/giflib-5.2.1.tar.gz
==> Patching
patching file Makefile
==> make all
gcc-4.2 -std=gnu99 -fPIC -Wall -Wno-format-truncation -O2   -c -o dgif_lib.o dgif_lib.c
cc1: error: unrecognized command line option "-Wno-format-truncation"
make: *** [dgif_lib.o] Error 1
==> Formula
Path: /usr/local/Library/Formula/giflib.rb
==> Configuration
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/mistydemeo/tigerbrew.git
HEAD: 05210ee133bac70eb23cd3626263c9185d5bb9c4
Last commit: 20 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://ia904500.us.archive.org/24/items/tigerbrew
CPU: dual-core 64-bit g5
OS X: 10.5.9-Power Macintosh
Curl: /usr/local/Library/Homebrew/vendor/portable-curl/current/bin/curl
Xcode: 3.1.4
CLT: N/A
GCC-4.0: build 5493
GCC-4.2: build 5577
LLVM-GCC: build 2064
Clang: N/A
X11: 2.1.6 => /usr/X11
System Ruby: 2.3.3-222 => /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.18_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/bin/ruby
Java: 1.5.0_30-b03
==> ENV
HOMEBREW_CC: gcc-4.2
HOMEBREW_CXX: g++-4.2
MAKEFLAGS: -j1
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.5
HOMEBREW_MAKE_JOBS: 1
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Library/ENV/4.3:/usr/bin:/bin:/usr/sbin:/sbin

Error: giflib 5.2.1 did not build
Logs:
     /Users/brian/Library/Logs/Homebrew/giflib/01.make
     /Users/brian/Library/Logs/Homebrew/giflib/01.make.cc

/usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/net/http.rb:933:in `connect_nonblock': Failed to connect to: https://api.github.com/search/issues?q=giflib+repo:mistydemeo/tigerbrew+in:title+state:open&per_page=100 (GitHub::Error)
SSL_connect returned=1 errno=0 state=error: certificate verify failed
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/net/http.rb:933:in `connect'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/net/http.rb:863:in `do_start'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/net/http.rb:852:in `start'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:319:in `open_http'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:737:in `buffer_open'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:212:in `block in open_loop'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:210:in `catch'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:210:in `open_loop'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:151:in `open_uri'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:717:in `open'
	from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/open-uri.rb:31:in `open'
	from /usr/local/Library/Homebrew/utils.rb:435:in `open'
	from /usr/local/Library/Homebrew/utils.rb:465:in `issues_matching'
	from /usr/local/Library/Homebrew/utils.rb:497:in `issues_for_formula'
	from /usr/local/Library/Homebrew/exceptions.rb:209:in `fetch_issues'
	from /usr/local/Library/Homebrew/exceptions.rb:205:in `issues'
	from /usr/local/Library/Homebrew/exceptions.rb:248:in `dump'
	from /usr/local/Library/brew.rb:170:in `rescue in <main>'
	from /usr/local/Library/brew.rb:61:in `<main>'
➜  ~

➜  giflib cat 01.make
2023-11-22 21:35:18 -0800

make
all

gcc-4.2 -std=gnu99 -fPIC -Wall -Wno-format-truncation -O2   -c -o dgif_lib.o dgif_lib.c
cc1: error: unrecognized command line option "-Wno-format-truncation"
make: *** [dgif_lib.o] Error 1

HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/mistydemeo/tigerbrew.git
HEAD: 05210ee133bac70eb23cd3626263c9185d5bb9c4
Last commit: 20 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://ia904500.us.archive.org/24/items/tigerbrew
CPU: dual-core 64-bit g5
OS X: 10.5.9-Power Macintosh
Curl: /usr/local/Library/Homebrew/vendor/portable-curl/current/bin/curl
Xcode: 3.1.4
CLT: N/A
GCC-4.0: build 5493
GCC-4.2: build 5577
LLVM-GCC: build 2064
Clang: N/A
X11: 2.1.6 => /usr/X11
System Ruby: 2.3.3-222 => /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Perl: /usr/bin/perl
Python: /usr/bin/python => /System/Library/Frameworks/Python.framework/Versions/2.5/Resources/Python.app/Contents/MacOS/Python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/bin/ruby
Java: 1.5.0_30-b03

HOMEBREW_CC: gcc-4.2
HOMEBREW_CXX: g++-4.2
MAKEFLAGS: -j1
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.5
HOMEBREW_MAKE_JOBS: 1
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Library/ENV/4.3:/usr/bin:/bin:/usr/sbin:/sbin
➜  giflib

➜  ~ brew doctor
Please note that these warnings are just used to help the Tigerbrew maintainers
with debugging if you file an issue. If everything you use Tigerbrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Some installed formula are missing dependencies.
You should `brew install` the missing dependencies:

    brew install giflib

Run `brew missing` for more details.
➜  ~

--- a/Library/Formula/giflib.rb
+++ b/Library/Formula/giflib.rb
@@ -19,7 +19,9 @@ class Giflib < Formula
     ENV.universal_binary if build.universal?
     ENV.enable_warnings if ENV.compiler == :gcc_4_0
 
-    system "make", "all"
+    # Pass CFLAGS to explicitly override the CFLAGS from the makefile,
+    # which contains some flags that are incompatible with GCC 4.0/4.2
+    system "make", "all", "CFLAGS=#{ENV.cflags}"
     system "make", "install", "PREFIX=#{prefix}"
   end
 

